### DISCUSSION TITLE
TCGA - Including copy number data into oncoplots

### DISCUSSION THREAD

**Discussion by LeaLe88** [PoisonAlien/maftools#985] [Q&A]
Good afternoon,

I have been trying to include copy number data into oncoplots following this instructions (section 0.3.2 (Custum copy-number table)) since I do not have GISTIC results for the TCGA KIRC cohort.

https://www.bioconductor.org/packages/devel/bioc/vignettes/maftools/inst/doc/oncoplots.html#032_Custom_copy-number_table

However, in the last step I keep getting the following error:
```r
kirc.plus.cn = read.maf(maf = kirc,
                       cnTable = custom.cn.data,
                         verbose = FALSE) 
Error in file.info(file) : invalid filename argument
```

Has anyone an idea what I can do?

The same issue was described here: https://github.com/PoisonAlien/maftools/issues/871

Previously I have downloaded the TCGA KIRC data and have KIRC_TCGA_gene-level-cnv.RDS, KIRC_TCGA_snv.maf, KIRC_TCGA_snv.maf.RDS, etc. available.

Thank you !

**Comment by PoisonAlien** [comment]
Hi,

Is `custom.cn.data` a data.frame or a file path?

**Reply by LeaLe88** [reply]
I tried both as I read somewhere that it should be a file path:
As dataframe:
kirc.plus.cn = read.maf(maf = kirc,
                       cnTable = custom.cn.data,
                         verbose = FALSE) 

With Path:
kirc.plus.cn = read.maf(maf = kirc,
                        cnTable = "/...path.../custom.cn.data.txt",verbose=FALSE)

**Reply by PoisonAlien** [reply]
I am having a hard time reproducing the issue. It works fine with input as a data.frame and a file. 
Could you please share the `custom.cn.data.txt`?

**Comment by LeaLe88** [comment]
Sure, I only made it for one gene as a test. 
[custom.cn.data.txt](https://github.com/PoisonAlien/maftools/files/13251799/custom.cn.data.txt)

**Reply by PoisonAlien** [reply]
Hi,
I have no problem working with your file.

```r
laml.maf = system.file("extdata", "tcga_laml.maf.gz", package = "maftools") #MAF file

m <- maftools::read.maf(maf = laml.maf, cnTable = "~/Downloads/custom.cn.data.txt")
-Reading
-Validating
-Silent variants: 475 
-Processing copy number data
-Summarizing
-Processing clinical data
--Missing clinical data
-Finished in 0.224s elapsed (0.210s cpu) 
```

Could you please try to update your package and try again?

**Reply by LeaLe88** [reply]
Thank you. Is it possible it has to do with my maf file?  I had a feeling this could be the cause maybe. I will update the package however.
[Uploading rstudio-export.zip…]()

**Reply by PoisonAlien** [reply]
Can you read the maf file alone without the copy number data?

**Reply by LeaLe88** [reply]
Yes it worked without it.

**Reply by LeaLe88** [reply]
I am trying to load my MAF file in above comment in zip, but I think it is not uploading

**Reply by LeaLe88** [reply]
any more ideas? thank you!

**Reply by PoisonAlien** [reply]
Could you please [email](mailto:anandmt3@gmail.com) me a reproducible MAF if possible? You can remove all sensitive information and change sample names if you'd like.

**Reply by LeaLe88** [reply]
Thank you so much for your reply. I will email it to you now. It is from TCGA, so no problem with sensitive information.

**Reply by PoisonAlien** [reply]
Hi,

Thank you for the data. I will answer here so that we can keep things in the chain. I tried with your data and it went through normally without any errors.

```r
> m = maftools::read.maf(maf = "~/Downloads/rstudio-export/KIRC_TCGA_filtered_snv.maf", cnTable = "~/Downloads/rstudio-export/custom.cn.data.txt")
-Reading
-Validating
-Silent variants: 5150 
-Processing copy number data
-Summarizing
--Mutiple centers found
BI;BCM;BCM;BI--Possible FLAGS among top ten genes:
  TTN
  MUC16
-Processing clinical data
--Missing clinical data
-Finished in 1.941s elapsed (1.618s cpu) 

> m
An object of class  MAF 
                        ID       summary        Mean Median
 1:             NCBI_Build        GRCh38          NA     NA
 2:                 Center BI;BCM;BCM;BI          NA     NA
 3:                Samples           351          NA     NA
 4:                 nGenes          9128          NA     NA
 5:                DeepDel            86  0.24501425      0
 6:        Frame_Shift_Del          1710  4.87179487      4
 7:        Frame_Shift_Ins           506  1.44159544      1
 8:           In_Frame_Del           248  0.70655271      0
 9:           In_Frame_Ins            21  0.05982906      0
10:      Missense_Mutation         13379 38.11680912     34
11:      Nonsense_Mutation          1010  2.87749288      2
12:       Nonstop_Mutation            24  0.06837607      0
13:             ShallowAmp            96  0.27350427      0
14:            Splice_Site           468  1.33333333      1
15: Translation_Start_Site            27  0.07692308      0
16:                  total         17575 50.07122507     46
17:                    Amp            77  0.21937322      0
18:                    Del            92  0.26210826      0
19:              CNV_total           169  0.48148148      0
```

Maybe you could try updating the package?

**Comment by LeaLe88** [comment]
Good afternoon,

Thank you so much for sharing your code adding "maftools::" in front of
read.maf actually solved my problem!!
I have loaded it now. I hope you dont mind if I ask you two additional
questions which I cant find an answer to.

1) It was described under 0.3.2 (
https://www.bioconductor.org/packages/devel/bioc/vignettes/maftools/inst/doc/oncoplots.html#032_Custom_copy-number_table)
that a "*random* CN status for above samples" is created. Now I am
wondering if this example is only for demonstration and I cant not actually
use the created copy-number table for my *actual analysis*?

2) I would like to combine my newly
 KIRC_TCGA_filtered_met450.RDS.zip
<https://drive.google.com/file/d/1MLRYsZtVM3j0s5DoV7yPVbKwc7A40dzb/view?usp=drive_web>
created MAF file also with the methylation data from TCGA (please find the
file attached). Do you have any idea how I can merge them?


Thank you again for your help, very appreciated.

Best Regards,
Lea

**Comment by PoisonAlien** [comment]
Hi,

I am glad it worked. It seems that you forgot to load the library before reading the MAF file. 

1. Yes, you can use your file as a copy number source. The vignette shows dummy data just as an example. 
2. It is not possible to merge methylation data with MAF.

**Comment by LeaLe88** [comment]
Hello,

Thank you for your response.
Regarding 2 however I saw TCGA oncoplots, which contained mutations,
deletions and methylation data. I assume that somehow the data was merged.
Thank you anyway!

Best Regards,
Lea
